Services/Diagnostics/DiagnosticsLogger.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Http;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics
+{
+    public interface IDiagnosticsLogger
+    {
+        Task<IHttpResponse> LogServiceStartAsync(string message);
+
+        Task<IHttpResponse> LogServiceHeartbeatAsync();
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            string exceptionMessage = "",
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            object data,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);        
+    }
+
+    public class DiagnosticsLogger : IDiagnosticsLogger
+    {
+        public struct JsonStruct
+        {
+            public string EventType;
+            public Dictionary<string, object> EventProperties;
+
+            public JsonStruct(string eventType, Dictionary<string, object> eventProps)
+            {
+                this.EventType = eventType;
+                this.EventProperties = eventProps;
+            }
+        }
+
+        private readonly IHttpClient httpClient;
+        private readonly IServicesConfig servicesConfig;
+        private readonly string diagnosticsEndpoint = string.Empty;
+
+        private const string SERVICE_ERROR_EVENT = "ServiceError";
+        private const string SERVICE_START_EVENT = "ServiceStart";
+        private const string SERVICE_HEARTBEAT_EVENT = "ServiceHeartbeat";
+        
+        public DiagnosticsLogger(IHttpClient httpClient, IServicesConfig servicesConfig)
+        {
+            this.httpClient = httpClient;
+            this.servicesConfig = servicesConfig;
+            this.diagnosticsEndpoint = this.servicesConfig.DiagnosticsEndpointUrl + "/diagnosticsevents";
```

Comment:
can we move this to the configuration file? e.g. `diagnostics_endpoint_url = "${?PCS_DIAGNOSTICS_WEBSERVICE_URL}/diagnosticsevents"`.  I rather not hardcode this, so it's easier to swap the service out with another one that used a different URL